From 145041ec75eae1e58fcd3f784964303fb75c16d5 Mon Sep 17 00:00:00 2001
From: Augusto Caringi <acaringi@redhat.com>
Date: Wed, 5 Feb 2020 15:24:21 +0100
Subject: [PATCH] Add missing <string> include to btf.h

The missing include was causing building problems with gcc 10.0.1:

In file included from
/builddir/build/BUILD/bpftrace-0.9.4/src/btf.cpp:1:
/builddir/build/BUILD/bpftrace-0.9.4/src/btf.h:26:8: error: 'string' in
namespace 'std' does not name a type
   26 |   std::string c_def(std::unordered_set<std::string>& set);
      |        ^~~~~~
/builddir/build/BUILD/bpftrace-0.9.4/src/btf.h:7:1: note: 'std::string'
is defined in header '<string>'; did you forget to '#include <string>'?
    6 | #include <unordered_set>
  +++ |+#include <string>
    7 |
---
 src/btf.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/btf.h b/src/btf.h
index 05b56ff..8856bfc 100644
--- a/src/btf.h
+++ b/src/btf.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <linux/types.h>
+#include <string>
 #include <unistd.h>
 #include <unordered_set>
 
-- 
2.21.1

